x/tx/signing/aminojson/encoder.go

```diff
@@ -95,7 +96,11 @@ func cosmosInlineJSON(_ *Encoder, v protoreflect.Value, w io.Writer) error {
 		if !json.Valid(bz) {
 			return errors.New("invalid JSON bytes")
 		}
-		_, err := w.Write(bz)
+		json, err := sortedJsonStringify(bz)
+		if err != nil {
+			return errors.Wrap(err, "invalid JSON bytes")
```

Comment:
This can also be a different error: `json.Marshal(sorted)` (not sure if practically possible though). Maybe better

```suggestion
			return errors.Wrap(err, "could not normalize JSON")
```